src/Pages/Profile/admin/AdminView.jsx

```diff
@@ -0,0 +1,355 @@
+  const [firstName, setFirstName] = useState("")
+  const [lastName, setLastName] = useState("")
+  const [middleInitial, setMiddleInitial] = useState("")
+  const [email, setEmail] = useState("")
+  const [password, setPassword] = useState("")
+  const [doorCode, setDoorCode] = useState("")
+  const [user, setUser] = useState({ ...props.user })
+  const [toggle, setToggle] = useState(false)
+  const [pagesPrinted, setPagesPrinted] = useState(user.pagesPrinted)
+  const [toggleSubmit, setToggleSubmit] = useState(false)
+  const [userMembership, setuserMembership] = useState(user.accessLevel)
+  const [numberOfSemestersToSignUpFor, setNumberOfSemestersToSignUpFor] = useState(undefined)
+  const [membershipValidUntil, setMembershipValidUntil] = useState(user.membershipValidUntil)
+
+  function handleToggle() {
+    setToggle(!toggle)
+  }
+
+  async function handleSubmission() {
+    const queryEmail = user.email
+
+    //hash pass
+    const salt = bcrypt.genSaltSync(10)
+    const hashed = (password.trim()==='') ?
+    user.password : bcrypt.hashSync(password, salt)
+
+    //Change term?
+    numberOfSemestersToSignUpFor===0 &&
+    setNumberOfSemestersToSignUpFor(undefined)
+
+    const editedUser = {
+      ...user,
+      firstName: firstName || user.firstName,
+      lastName: lastName || user.lastName,
+      middleInitial: middleInitial || user.middleInitial,
+      email: email || user.email,
+      password: hashed,
+      doorCode: doorCode || user.doorCode,
+      pagesPrinted: pagesPrinted,
+      accessLevel: userMembership,
+      numberOfSemestersToSignUpFor: numberOfSemestersToSignUpFor
+    }
+
+    setUser({...editedUser})
+    setToggle(!toggle)
+
+    await axios
+      // get all user!
+      .post('/api/user/edit', {
+        ...editedUser,
+        queryEmail: queryEmail,
+        token: props.token
+      })
+      .then(result => {
+        if (result.status >= 200 && result.status < 300) {
+          setMembershipValidUntil(result.data.membershipValidUntil)
+        }
+      })
+      .catch(err => {
+        console.log(err)
+      })
+
+    setToggle(false)
+    setToggleSubmit(false)
+  }
+
+  function handleSubmissionToggle() {
+    if (user.email === email)
+      setToggleSubmit(!toggleSubmit)
+    else {
+      checkIfUserExists(email)
+    }
+  }
+
+  async function checkIfUserExists(val) {
+    await axios
+      .post('/api/user/checkIfUserExists', { email: val })
+      .then(result => {
+        if (result.status >= 200 && result.status < 300) {
+          setToggleSubmit(!toggleSubmit)
+        }
+      })
+      .catch(err => {
+        window.alert('The Email has already existed, leave it blank if you do not want to change')
+        console.log(err)
+      })
+  }
+
+  /*
+  translating access Level into role
+  The access level is defined as follows:
+  -2: Ban
+  -1: Pending
+   0: Member
+   1: Officer
+   2: Admin
+  */
+  function roleTranslator(level) {
```

Comment:
lets add the enums in this PR. this is duplicated twice across Overview and AdminView